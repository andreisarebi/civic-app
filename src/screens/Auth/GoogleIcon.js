import React from 'react';
import { Image, StyleSheet, View } from 'react-native';

const icon = require('../../assets/images/google-icon.png');

const GoogleIcon = () => (
  <View style={styles.iconContainer}>
    <Image source={icon} style={styles.googleIcon} />
  </View>
);

const styles = StyleSheet.create({
  googleIcon: {
    height: 32,
    width: 32,
  },
  iconContainer: {
    marginRight: 12,
  },
});

export default GoogleIcon;
