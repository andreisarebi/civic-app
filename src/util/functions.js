import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
