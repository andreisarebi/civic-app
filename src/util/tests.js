import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
