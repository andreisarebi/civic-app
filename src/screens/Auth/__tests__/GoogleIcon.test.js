import { createSnapshot } from '../../../util/tests';
import GoogleIcon from '../GoogleIcon';

describe('GoogleIcon', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(GoogleIcon);
    expect(snapshot).toMatchSnapshot();
  });
});
