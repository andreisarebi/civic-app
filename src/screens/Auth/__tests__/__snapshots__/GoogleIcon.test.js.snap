// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleIcon renders correctly 1`] = `
<Component
  style={
    Object {
      "marginRight": 12,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 32,
        "width": 32,
      }
    }
  />
</Component>
`;
